buildenv/jenkins/JenkinsfileBase

```diff
@@ -746,6 +752,25 @@ def post(output_name) {
 					uploadToArtifactory(pattern)
 				}				
 			}
+			if (currentBuild.result == 'UNSTABLE' || currentBuild.result == 'FAILURE' || currentBuild.result == 'ABORTED') {
+				echo "Fetching Tap files..."
+				def tapFiles = findFiles(glob: "**/*.tap")
+				for (def tapFile: tapFiles) {
+					echo "Tap file found: ${tapFile}"
+					def file = readFile(file: "${tapFile}")
+					TapConsumer tapConsumer = TapConsumerFactory.makeTap13Consumer()
+					TestSet testSet = tapConsumer.load(file)
+					for (int i = 1; i <= testSet.getNumberOfTestResults(); i++) {
+						if (!testSet.getTestResult(i).getStatus().toString().equals("ok")) {
+							def testCase = testSet.getTestResult(i).getDescription().substring(2)
+							env.FAIL_TESTLIST += "$testCase,"
+						}
+					}
+				}
+				env.FAIL_TESTLIST = env.FAIL_TESTLIST.substring(0, env.FAIL_TESTLIST.length() - 1)
+				echo "Fail test list is ${FAIL_TESTLIST}"
+				addFailTestsGrinderLink()
+			}
```

Comment:
- please use a function, so we do not have to duplicate the code at line 1044
- please use a local variable for the failed test list (not environment variable) - `env.FAIL_TESTLIST` (please also rename as suggested by Shelley)

Reply:
After tidying the code, the reruning Grinder link is [Grinder 3776](https://ci.adoptopenjdk.net/job/Grinder/3776/)